scipy/signal/tests/test_ltisys.py

```diff
@@ -8,11 +8,62 @@
 from scipy.signal.ltisys import (ss2tf, tf2ss, lsim2, impulse2, step2, lti,
                                  bode, freqresp, lsim, impulse, step,
                                  abcd_normalize, place_poles,
-                                 TransferFunction, StateSpace, ZerosPolesGain)
+                                 TransferFunction, StateSpace, ZerosPolesGain,
+                                 controllability_matrix,
+                                 observability_matrix, controller_hessenberg)
+
 from scipy.signal.filter_design import BadCoefficients
 import scipy.linalg as linalg
 
-
+class TestCtrbObsv:    
+    def test_hessenberg(self):
+        A = np.array([[1,1,0],[0,1,0],[0,0,1]])
+        # this system is not controllable
+        B = np.array([[0,0],[1,0],[1,0]])
+        H,Bbar,Cbar,P,c_idx = controller_hessenberg(A,B)
+        assert_equal(2,c_idx)
+        assert np.allclose(P.dot(A.dot(P.T)), H)
+        assert np.allclose(P.dot(B), Bbar)
+        assert np.allclose(np.linalg.inv(P), P.T)
```

Comment:
No plain ``assert``s should be used in testcases or code, because those are stripped when byte-compiling with ``-O``.

In this case, use ``assert_allclose`` instead of ``assert np.allclose``.